src/main/kotlin/yjh/cstar/engine/application/GameEngineService.kt

```diff
@@ -59,25 +63,23 @@ class GameEngineService(
                 logger.info { "[WARN] busy waiting..." }
                 gameAnswerPollRepository.poll(roomId, quiz.id)
                     ?.takeIf {
-                        it.answer.replace(" ", "")
-                            .equals(quiz.answer.replace(" ", ""), ignoreCase = true)
+                        answerValidationService.validateAnswer(it.answer, quiz.answer)
                     }
                     ?.let { result ->
-                        updateScore(ranking, result.playerId)
+                        scoreManagementService.updateScore(ranking, result.playerId)
                         nicknames[result.playerId] = result.nickname
                         broadcastResult(destination, result)
-                        broadcastRanking(ranking, nicknames, destination)
+                        val rankingMessage = rankingService.generateRanking(ranking, nicknames)
+                        broadcastRanking(destination, rankingMessage)
```

Comment:
RankingService도 위의 ScoreService와 마찬가지로 @Service가 아닌 일반 도메인 클래스로 만들어보면 좋을 것 같아요!!

필드로는 변경가능한 ranking필드를 가지고 있고, Ranking이라는 클래스에게 메세지를 보내는 형태로요!!

```suggestion
class Ranking(
    val ranking = TreeMap<Long, Int>()
) { ... 외부에서 호출할 메서드 정의 }
```

Ranking산정 방식이 복잡해질 것 같다면 RankingService는 Bean으로 유지하고, RankingService 내부에서 Ranking이라는 객체를 던져주면 요리(?)하는 식으로 구현해봐도 좋을 것 같습니다!!

Reply:
두가지 방식 다 좋을 것 같습니다. 직접 리팩토링하면서 적용해보고 올려보겠습니다!